openverse_catalog/dags/database/report_pending_reported_media.py

```diff
@@ -0,0 +1,171 @@
+
+An image may be reported for mature content or copyright infringement, for
+example. Once reported, these require manual review through the Django Admin to
+determine whether further action (such as deindexing the record) needs to be
+taken. If an image has been reported multiple times, it only needs to be
+reviewed once and so is only counted once in the reporting by this DAG.
+
+The DAG should be run weekly.
+"""
+import logging
+import os
+from textwrap import dedent
+from urllib.parse import urljoin
+
+from airflow import DAG
+from airflow.exceptions import AirflowException
+from airflow.models import TaskInstance
+from airflow.operators.python import PythonOperator
+from airflow.providers.postgres.hooks.postgres import PostgresHook
+from common import slack
+from common.constants import DAG_DEFAULT_ARGS, MEDIA_TYPES, XCOM_PULL_TEMPLATE
+
+
+logger = logging.getLogger(__name__)
+
+DAG_ID = "report_pending_reported_media"
+DB_CONN_ID = os.getenv("OPENLEDGER_API_CONN_ID", "postgres_openledger_api")
+
+REPORTS_TABLES = {
+    "image": "nsfw_reports",
+    "audio": "nsfw_reports_audio",
+}
+
+# Column name constants
+URL = "identifier"
+STATUS = "status"
+REASON = "reason"
+
+ReportCountsByReason = dict[str, int]
+MediaTypeReportCounts = dict[str, ReportCountsByReason]
+
+
+def get_pending_report_counts(
+    db_conn_id: str, media_type: str, ti: TaskInstance
+) -> ReportCountsByReason:
+    """
+    For the given media type, builds a dict of pending report counts grouped by
+    report reason.
+
+    Required Arguments:
+
+    db_conn_id: Connection ID for the API database
+    media_type: Media type for which to look up reports
+    ti:         Running task instance
+    """
+    postgres = PostgresHook(postgres_conn_id=db_conn_id)
+    report_counts_by_reason = {}
+
+    if media_type not in REPORTS_TABLES:
+        raise AirflowException(f"{media_type} has no Reports table configured.")
+
+    query = dedent(
+        f"""
+        SELECT
+        COUNT(DISTINCT {URL}), {REASON}
+        FROM {REPORTS_TABLES[media_type]}
+        WHERE {STATUS}='pending_review'
+        GROUP BY {REASON}
+        """
+    )
+    for count, reason in postgres.get_records(query):
+        report_counts_by_reason[reason] = count
+
+    logger.info(f"{report_counts_by_reason} records require manual review")
+
+    return report_counts_by_reason
+
+
+def report_actionable_records(
+    report_counts_by_media_type: MediaTypeReportCounts,
+) -> None:
+    """
+    Build a report detailing pending reports for all media types, and alert to Slack.
+
+    Required Arguments:
+
+    report_counts_by_media_type: dict containing report counts per media type, per
+                                 report reason
+    """
+    if all(len(reports) == 0 for reports in report_counts_by_media_type.values()):
+        slack.send_message(
+            "No records require review at this time :tada:",
+            username="Reported Media Check-In",
+        )
+        return
```

Comment:
For me it feels useful to have a weekly peace-of-mind status report regardless -- this just eliminates the situation where you feel like you haven't heard from Reported Media in awhile and must go manually check DAG logs. I do understand the concern about notification fatigue, but this would be once weekly in the crop of notifications that come in over the weekend.

I don't feel especially strongly about this though! It seems likely that these notifications may change as we update our processes for dealing with reported media at any rate 😄